core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -304,8 +304,8 @@ void assertUnsupportedIndexHandling(String index) throws Exception {
         try {
             client().admin().indices().prepareOpen(indexName).get();
             fail("Shouldn't be able to open an old index");
-        } catch (IndexException ex) {
-            assertThat(ex.getMessage(), containsString("cannot open the index due to upgrade failure"));
+        } catch (IllegalStateException ex) {
+            assertThat(ex.getMessage(), containsString("was created before v0.90.0 and wasn't upgraded"));
```

Comment:
different change - but I think we should have the 0.90.0 as Version.minimumDataCompatibilityVersion ..